### PR TITLE
fix: update to new repos

### PR DESCRIPTION
**Pull request by JacobCoffee**
## Description

- Attempts to update to [new Salt project repos](https://saltproject.io/blog/salt-project-package-repo-migration-and-guidance/)
- 
Notably, versions prior 3006 are *completely* removed
```
Decommissioning of Older Packages

As part of this migration, Salt Project is dropping all packages older than Salt v3006. This means that Salt 3005, 3004, etc. and older will no longer be available for installation.

At the end of October 2024, packages will no longer be available via:

    repo.saltproject.io
    archive.repo.saltproject.io
```

### CODE REVIEW

File: dockerfiles/Dockerfile.jammy [hunk 1]
```diff
@@ -47,8 +47,13 @@ EXPOSE 22
...
 
 # Setup Salt Common
-RUN wget --quiet -O /etc/apt/keyrings/salt-archive-keyring-2023.gpg https://repo.saltproject.io/salt/py3/ubuntu/22.04/$(dpkg --print-architecture)/SALT-PROJECT-GPG-PUBKEY-2023.gpg
-RUN echo "deb [signed-by=/etc/apt/keyrings/salt-archive-keyring-2023.gpg arch=$(dpkg --print-architecture)] https://repo.saltproject.io/salt/py3/ubuntu/22.04/$(dpkg --print-architecture)/3006 jammy main" > /etc/apt/sources.list.d/salt.list
+RUN mkdir -p /etc/apt/keyrings
+RUN wget --quiet -O /etc/apt/keyrings/salt-archive-keyring-2024.pgp https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public
+RUN echo "deb [signed-by=/etc/apt/keyrings/salt-archive-keyring-2024.pgp arch=$(dpkg --print-architecture)] https://packages.broadcom.com/artifactory/saltproject-deb/ stable main" > /etc/apt/sources.list.d/salt.list
+
+# Pin to Salt 3006 LTS
+RUN echo -e "Package: salt-*\nPin: version 3006.*\nPin-Priority: 1001" > /etc/apt/preferences.d/salt-pin-1001
```
Comment (JacobCoffee) [inline dockerfiles/Dockerfile.jammy:55]: not quite working

**Comment by JacobCoffee** [comment]
- what does this mean to our lingering friends pre-jammy? we can't really do conditionals *at all* because the repos are just... gone

File: docs/guides/migration-recipe.md [hunk 2]
```diff
@@ -99,9 +99,9 @@ index 68387c9..7a8ace1 100644
     CODENAME=$(cat /etc/os-release | grep VERSION_CODENAME | cut -d '=' -f 2)
    
     echo "Adding the SaltStack repository key for $UBUNTU_VERSION $CODENAME ($ARCH)..."
-    sudo curl -fsSL -o /etc/apt/keyrings/salt-archive-keyring-2023.gpg https://repo.saltproject.io/salt/py3/ubuntu/$UBUNTU_VERSION/$ARCH/SALT-PROJECT-GPG-PUBKEY-2023.gpg
+    sudo curl -fsSL -o /etc/apt/keyrings/salt-archive-keyring-2024.gpg ttps://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public
```
Comment (ewdurbin) [inline docs/guides/migration-recipe.md:102]:
```suggestion
    sudo curl -fsSL -o /etc/apt/keyrings/salt-archive-keyring-2024.gpg https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public
```

**Review by ewdurbin — COMMENTED** [review]
https://docs.saltproject.io/salt/install-guide/en/latest/topics/salt-supported-operating-systems.html#salt-supported-operating-systems indicates that 20.04 is still supported for 3006, should we try that for `focal`?

**Comment by JacobCoffee** [comment]
Tested focal for `pythontest` and it passed
```
    pythontest: 
    pythontest: Summary for local
    pythontest: --------------
    pythontest: Succeeded: 163
    pythontest: Failed:      0
    pythontest: --------------
    pythontest: Total states run:     163
    pythontest: Total run time:     8.186 s
```

File: Vagrantfile [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 
 SERVERS = [
+  # TODO: test focal salt repo changes
```
Comment (ewdurbin) [inline Vagrantfile:5]: Is this still TODO?

File: dockerfiles/Dockerfile.focal [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Docker image to use with Vagrant
+           # Docker image to use with Vagrant
```
Comment (ewdurbin) [inline dockerfiles/Dockerfile.focal:1]: can we remove this unnecessary indent?

File: docs/guides/migration-recipe.md [hunk 5]
```diff
@@ -99,9 +99,9 @@ index 68387c9..7a8ace1 100644
     CODENAME=$(cat /etc/os-release | grep VERSION_CODENAME | cut -d '=' -f 2)
    
     echo "Adding the SaltStack repository key for $UBUNTU_VERSION $CODENAME ($ARCH)..."
-    sudo curl -fsSL -o /etc/apt/keyrings/salt-archive-keyring-2023.gpg https://repo.saltproject.io/salt/py3/ubuntu/$UBUNTU_VERSION/$ARCH/SALT-PROJECT-GPG-PUBKEY-2023.gpg
+    sudo curl -fsSL -o /etc/apt/keyrings/salt-archive-keyring-2024.gpg https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public
     echo "Adding the SaltStack repository for $UBUNTU_VERSION $CODENAME ($ARCH)..."
-    echo "deb [signed-by=/etc/apt/keyrings/salt-archive-keyring-2023.gpg arch=$ARCH] https://repo.saltproject.io/salt/py3/ubuntu/$UBUNTU_VERSION/$ARCH/latest $CODENAME main" | sudo tee /etc/apt/sources.list.d/salt.list
+    echo "deb [signed-by=/etc/apt/keyrings/salt-archive-keyring-2024.gpg arch=$ARCH] https://packages.broadcom.com/artifactory/saltproject-deb/ stable main" | sudo tee /etc/apt/sources.list.d/salt.list
```
Comment (ewdurbin) [inline docs/guides/migration-recipe.md:104]: Should the pinning command be part of this doc now also?

File: salt/base/salt.sls [hunk 6]
```diff
@@ -35,23 +35,12 @@ remove_old_salt_repo:
...
   pkgrepo.managed:
-    - humanname: repo.saltstack.org
-    {% if grains["oscodename"] == "focal" %}
-    - name: deb https://archive.repo.saltproject.io/py3/ubuntu/20.04/{{ grains["osarch"] }}/archive/3004 focal main
-    - key_url: https://archive.repo.saltproject.io/py3/ubuntu/20.04/{{ grains["osarch"] }}/archive/3004/salt-archive-keyring.gpg
-    {% elif grains["oscodename"] == "jammy" %}
-    - name: deb [signed-by=/etc/apt/keyrings/salt-archive-keyring.gpg arch={{ grains["osarch"] }}] https://repo.saltproject.io/salt/py3/ubuntu/22.04/{{ grains["osarch"] }}/3007 jammy main
-    - key_url: https://repo.saltproject.io/salt/py3/ubuntu/22.04/{{ grains["osarch"] }}/SALT-PROJECT-GPG-PUBKEY-2023.gpg
+    {% if grains["oscodename"] in ["focal", "jammy", "noble"] %}
+    - name: deb [signed-by=/etc/apt/keyrings/salt-archive-keyring-2024.pgp arch={{ grains["osarch"] }}] https://packages.broadcom.com/artifactory/saltproject-deb/ stable main
```
Comment (ewdurbin) [inline salt/base/salt.sls:39]: Should we be doing something to pin the version of salt here?

Comment (JacobCoffee) [reply salt/base/salt.sls:52]: [1050e3a](https://github.com/python/psf-salt/pull/524/commits/1050e3a0dd8cb0f3cf8407622c32d29a1adf6e6c)

File: docs/guides/migration-recipe.md (hunk 5 above)
Comment (JacobCoffee) [reply docs/guides/migration-recipe.md:104]: [9acd4ba](https://github.com/python/psf-salt/pull/524/commits/9acd4bae92be781af8bbcff78296de87c447a52b)

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**